[Qin] iP

wangqinNick opened the pull request.

Commit:
Add Gradle support

wangqinNick committed:
Add Level 0. Greet

wangqinNick committed:
Add new entry to the main function

wangqinNick committed:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606'

* commit '3b19ba1790d6035a9e246f5d4ae627bf68778606':
  Add Gradle support

wangqinNick committed:
Add jansi library with Gradle plugin

wangqinNick committed:
Add Level-1: Greet, Echo, Exit

wangqinNick committed:
Restructure the whole project
Encapsulate most of the class and their dependencies

wangqinNick committed:
Add Level-2: Add, List

wangqinNick committed:
Add Level 3. Mark as Done

wangqinNick committed:
Clean up code
(Remove redundant. Make most things static as possible)

wangqinNick committed:
Edit ExitCommand taking use of System.exit(0)

wangqinNick committed:
Add Executor class (incomplete)

wangqinNick committed:
Add more to Executor class (incomplete)

wangqinNick committed:
Add todo task

wangqinNick committed:
Add deadline task (all support)

wangqinNick committed:
Add event class (full support)

wangqinNick committed:
Add A-TextUiTesting (build, run through jar file)

wangqinNick committed:
Debug: solve the warning of newLine

nat-ho commented on src/main/java/seedu/duck/ui/TextUi.java:
```diff
@@ -0,0 +1,199 @@
+                index,
+                eventTask.getType(),
+                eventTask.getChar(),
+                eventTask.getDescription(),
```
I like how you applied SLAP principles in your code. There is little to no varying levels of abstraction throughout your code within each fragment making it extremely easy to read and understand!

ychong032 commented on src/main/java/seedu/duck/util/Message.java:
```diff
@@ -0,0 +1,39 @@
+    public static final String MESSAGE_DUPLICATE_TASK_ALERT_2 = "Do you want to add a duplicate task?";
+    public static final String MESSAGE_DUPLICATE_TASK_ALERT_3 = "Press Y to add and others to not add: ";
+    public static final String MESSAGE_DUPLICATE_TASK_NOT_ADDED = "The duplicate task is not added!";
+}
```
I like the idea of storing all the print messages as constants in a separate class!

ychong032 commented on src/main/java/seedu/duck/system/TaskManager.java:
```diff
@@ -0,0 +1,57 @@
+        }
+    }
+
+    public static int size(){
```
Should this method be renamed to be a verb? E.g. getSize().

ychong032 reviewed (commented):
Overall, minimal code violations spotted. Your code is an excellent example of good coding standards, especially in terms of SLAP. Well done!

wangqinNick committed:
Add DukeException; Handle several exceptions

wangqinNick committed:
Merge branch 'branch-Level-5'

* branch-Level-5:
  Add DukeException; Handle several exceptions

wangqinNick committed:
Add interface Timeliness

wangqinNick committed:
Add JavaFx dependencies; Add Application class

wangqinNick committed:
Add designed layout and styled controls for GUI

wangqinNick committed:
Add fully functional GUI (Echoing)
Iteration 1 – Echoing the user
Iteration 2 – Adding Dialog Boxes
Iteration 3 – Adding custom behaviour to DialogBox

wangqinNick committed:
Add background image (does not look very good)

nat-ho commented on src/main/java/seedu/duck/command/add/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,39 @@
+        //according to the data format
+        return new CommandResult(
+                String.format(
+                    MESSAGE_DEADLINE_SUCCESS ,
```
Excellent use of named constant variables to avoid magic numbers making it a breeze to read. Additionally the entire code is structured very well and on top of that, all the 'happy paths' are prominent. Very well done overall!

wangqinNick committed:
Move further on GUI (delete unrelated codes)

wangqinNick committed:
Adjust the size of stage, scrollPane

wangqinNick committed:
Add adjustable dialog box

wangqinNick committed:
Add Background picture (perfect)

wangqinNick committed:
Add background music

wangqinNick committed:
Refactor main stage functions

wangqinNick committed:
Clean up codes

wangqinNick committed:
Use GIF as user/duke dialog pictures

wangqinNick committed:
Try to use stream in TaskManger (not very good)

wangqinNick committed:
Clean up

wangqinNick committed:
Minor changes

wangqinNick committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Minor changes
  Clean up
  Try to use stream in TaskManger (not very good)
  Use GIF as user/duke dialog pictures
  Clean up codes
  Refactor main stage functions
  Add background music
  Add Background picture (perfect)
  Add adjustable dialog box
  Adjust the size of stage, scrollPane
  Move further on GUI (delete unrelated codes)
  Add background image (does not look very good)
  Add fully functional GUI (Echoing) Iteration 1 – Echoing the user Iteration 2 – Adding Dialog Boxes Iteration 3 – Adding custom behaviour to DialogBox
  Add designed layout and styled controls for GUI
  Add JavaFx dependencies; Add Application class

wangqinNick committed:
Add DeleteCommand

wangqinNick committed:
Add IOmanager save (wrong location)

wangqinNick committed:
Handle folder not exist issue

wangqinNick committed:
Merge branch 'branch-Level-6'

* branch-Level-6:
  Add DeleteCommand

wangqinNick committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Handle folder not exist issue
  Add IOmanager save (wrong location)

wangqinNick committed:
Use separate JavaFx SDK, instead of the Intellij one

wangqinNick committed:
clean up

wangqinNick committed:
Remove potential bug part(bgm)

wangqinNick committed:
Refactor some codes

wangqinNick committed:
Teach Duke how to understand dates and times
Duke should understand 2/12/2019

wangqinNick committed:
Level 9. Find
Give users a way to find a task by searching for a keyword.

wangqinNick committed:
Update the taskManager

wangqinNick committed:
Update the Time Parser

wangqinNick committed:
Merge pull request #1 from wangqinNick/branch-Level-8

Branch level 8

wangqinNick committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level 9. Find Give users a way to find a task by searching for a keyword.

wangqinNick committed:
Update the IO loader

wangqinNick committed:
Add DateTime implementation v0.9

wangqinNick committed:
Add full support of DateTime

wangqinNick committed:
Add 1/3 JavaDoc

wangqinNick committed:
Clean up

wangqinNick committed:
Debug index

wangqinNick committed:
Add a different colour code for Invalid command response

wangqinNick committed:
Debug for Date and Time parser

wangqinNick committed:
Add different colour codes and prompt types for different commands

wangqinNick committed:
Add welcome message

wangqinNick committed:
Minor refctoring

wangqinNick committed:
Use Stream and Lambda

wangqinNick committed:
Use java var for local variables

wangqinNick committed:
Update the dependencies for javafx media

wangqinNick committed:
Fix negative index bug

wangqinNick committed:
Refactor the TaskManager (make it more static)

wangqinNick committed:
Add init function for DuckApp
(DoneCommand not working)

wangqinNick committed:
Add a pair of constructor for Todo, Deadline, Event task

wangqinNick committed:
Fix loadList can not load as done task bug

wangqinNick committed:
Add media support

wangqinNick committed:
Add undo command

wangqinNick committed:
Add saveState (fully support of UndoCommand)

wangqinNick committed:
Fix some indexing bugs

wangqinNick committed:
Add UndoCommand Usage in HelpCommand

wangqinNick committed:
Add DueCommand (find a task on specific date)

wangqinNick committed:
Use more java Streams and Lambdas

wangqinNick committed:
Clean up

wangqinNick committed:
Add another 1/3 JavaDoc

wangqinNick committed:
Add more JavaDoc

wangqinNick committed:
Add 100% JavaDoc

wangqinNick committed:
Add a duck picture

wangqinNick committed:
Adjust the colour code and tick/cross symbol

wangqinNick committed:
Change some typo

wangqinNick committed:
Finalise the javaDoc

wangqinNick committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Finalise the javaDoc

wangqinNick committed:
Update README.md

Test a README.md

wangqinNick committed:
Merge pull request #2 from wangqinNick/branch-UserGuide

Test a dummy README.md

wangqinNick committed:
Add a picture for README.md

wangqinNick committed:
Set theme jekyll-theme-minimal

wangqinNick committed:
Update README.md

wangqinNick committed:
Add  README.md

wangqinNick committed:
Merge branches 'master' and 'master' of https://github.com/wangqinNick/ip

* 'master' of https://github.com/wangqinNick/ip:
  Update README.md
  Set theme jekyll-theme-minimal

* 'master' of https://github.com/wangqinNick/ip:
  Update README.md
  Set theme jekyll-theme-minimal

wangqinNick committed:
Update README.md

wangqinNick committed:
Add Chinese support

wangqinNick committed:
Fix some bugs happened during language changing

wangqinNick committed:
Add JavaDoc for SystemSetting

wangqinNick committed:
Minor changes

wangqinNick committed:
Minor refectoring

wangqinNick committed:
Some useless changes in MediaPlayer

wangqinNick committed:
Add traverse in history command / CommandManager

wangqinNick committed:
Add minor changes

wangqinNick committed:
Add gif files for README.md

wangqinNick committed:
Move README.md resources to docs folder

wangqinNick committed:
Add README.md resources

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

xhxh96 commented on README.md:
````diff
@@ -1,26 +1,245 @@
-   | |_| | |_| |   <  __/
-   |____/ \__,_|_|\_\___|
-   ```
+# Duck - User Guide
````
Good job on updating the README!

xhxh96 commented on src/main/java/seedu/duck/command/DueCommand.java:
```diff
@@ -0,0 +1,85 @@
+import static seedu.duck.util.Message.MESSAGE_EMPTY_LIST_IN_CHINESE;
+import static seedu.duck.util.Message.MESSAGE_EMPTY_LIST_IN_ENGLISH;
+
+public class DueCommand extends Command{
```
Whitespace should be added after `Command`

xhxh96 commented on src/main/java/seedu/duck/command/DueCommand.java:
```diff
@@ -0,0 +1,85 @@
+     * Constructs the <b>Due Command</b>.
+     * <br> If the <code>specifier</code> given is <code>NULL</code>, the <code>timeSpecifier</code> is set to
+     * <i>on</i>.
+     * @param searchDate
+     *  The <i>date</i> to filter the <b>Task List</b> by
```
Description of `@param` should be on the same line as the parameter itself in Javadoc.

xhxh96 commented on src/main/java/seedu/duck/command/FindCommand.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public CommandResult executeInEnglish() {
+        filteredTaskList = TaskManager.searchTask(toSearch);
+        if (filteredTaskList.isEmpty()) return new CommandResult(MESSAGE_EMPTY_LIST);
```
Body of `if` statements should be wrapped in parenthesis and start on a new line after the `if` clause.

xhxh96 commented on src/main/java/seedu/duck/command/FindCommand.java:
```diff
@@ -0,0 +1,58 @@
+    public CommandResult executeInEnglish() {
+        filteredTaskList = TaskManager.searchTask(toSearch);
+        if (filteredTaskList.isEmpty()) return new CommandResult(MESSAGE_EMPTY_LIST);
+        var taskListMessage = getAppendedTasksMessage(filteredTaskList);
```
Try to use explicit type as much as possible in Java.

xhxh96 commented on src/main/java/seedu/duck/command/ListCommand.java:
```diff
@@ -0,0 +1,55 @@
+     */
+    @Override
+    public CommandResult executeInEnglish() {
+        if (TaskManager.getTaskList().size()>0){
```
Add whitespace between operators for greater readability.

xhxh96 commented on src/main/java/seedu/duck/command/PromptType.java:
```diff
@@ -0,0 +1,15 @@
+ * EDIT: the command is a EDIT command
+ * NONE: the command is a basic command
+ */
+public enum PromptType {
```
Good use of enumerations!

xhxh96 commented on src/main/java/seedu/duck/command/add/AddEventCommand.java:
```diff
@@ -0,0 +1,90 @@
+import seedu.duck.task.Task;
+
+import static seedu.duck.util.Constant.LIST_INDEX_OFFSET;
+import static seedu.duck.util.Message.*;
```
Avoid using wildcard operators when doing import.

xhxh96 commented on src/main/java/seedu/duck/data/CommandManager.java:
```diff
@@ -0,0 +1,37 @@
+        if (pointer <= 0){
+            //
+        }
+        pointer --;
```
Unnecessary whitespace between decrement operator and the variable.

xhxh96 commented on src/main/java/seedu/duck/storage/IOManager.java:
```diff
@@ -0,0 +1,117 @@
+        ArrayList<Task> tempTaskList = new ArrayList<>();
+        for (Task task : readList) {
+            switch (task.getType()) {
+            case 'T':
+                tempTaskList.add(new TodoTask(task.getDescription(), task.getIsDone()));
+                break;
+            case 'D':
+                if (task.getTaskDate()==null){
+                    tempTaskList.add(new DeadlineTask(task.getDescription(), task.getTaskDateInString(), task.getIsDone()));
+                }else {
+                    tempTaskList.add(new DeadlineTask(task.getDescription(), task.getTaskDate(), task.getIsDone()));
+                }
+                break;
+            case 'E':
```
You can consider using an enumeration for the task type rather than using magic character.

xhxh96 commented on src/main/java/seedu/duck/task/TaskType.java:
```diff
@@ -0,0 +1,7 @@
+package seedu.duck.task;
+
+public enum TaskType {
```
Oh I just realized you actually did declare an enumeration for this. You can use this in your `switch-case` statement above.

xhxh96 commented on src/main/java/seedu/duck/util/DateTime.java:
```diff
@@ -0,0 +1,238 @@
+     * @return The <code>date</code> in a string format.
+     */
+    private String dateToString() {
+        if (isToday()) {
+            return "today";
+        } else if (isTomorrow()) {
+            return "tomorrow";
```
Avoid magic string as much as possible.

xhxh96 reviewed (commented):
Good effort in doing up the UI and applying SLAP to your code! Left a few comments on some of the coding style violations, and consider specifying type when declaring variables/constant instead of using `var`. Otherwise code is clean and readable!

xhxh96 commented on src/main/java/seedu/duck/command/DeleteCommand.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public CommandResult executeInEnglish() {
+        try {
+            final var toDelete = getTargetTask();
```
Specifying the type is preferred over using type-inferred `var` type for variables/constants.

wangqinNick committed:
Set theme jekyll-theme-architect

wangqinNick committed:
Update doc/README.md

wangqinNick committed:
Merge branch 'master' of https://github.com/wangqinNick/ip

* 'master' of https://github.com/wangqinNick/ip:
  Set theme jekyll-theme-architect

wangqinNick committed:
Test

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Set theme jekyll-theme-architect

wangqinNick committed:
Update README.md

Update the path

wangqinNick committed:
clean up external libraries and resources

wangqinNick committed:
Change all wildcard imports to single import

wangqinNick committed:
Add handler to duplicated tasks

wangqinNick committed:
Add AllowDuplicatedTaskCommand

wangqinNick committed:
Minor changes

wangqinNick committed:
Add Background picture

wangqinNick committed:
implement scroll down / up

wangqinNick committed:
Minor changes

wangqinNick committed:
Add change username command / change password command / decoder class / encoder class

wangqinNick committed:
Add new task window

wangqinNick committed:
Add Menu Bar

wangqinNick committed:
Prepare necessary resources for README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Set theme jekyll-theme-cayman

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Add ui.gif

wangqinNick committed:
Merge branch 'master' of https://github.com/wangqinNick/ip

* 'master' of https://github.com/wangqinNick/ip:
  Update README.md
  Update README.md
  Set theme jekyll-theme-cayman
  Update README.md
  Update README.md
  Update README.md

wangqinNick committed:
Update README.md

wangqinNick committed:
Update README.md

Update the download link

wangqinNick committed:
Update README.md

wangqinNick committed:
Clean up
